include/misc/stack.h

```diff
@@ -90,8 +91,23 @@ static inline void stack_analyze(const char *name, const char *stack,
  * @param name Name of the stack
  * @param sym The symbol of the stack
  */
-#define STACK_ANALYZE(name, sym) \
-	stack_analyze(name, K_THREAD_STACK_BUFFER(sym), \
-		      K_THREAD_STACK_SIZEOF(sym))
+#if defined(CONFIG_INIT_STACKS)
+#define STACK_ANALYZE(name, sym)					\
+	do {								\
+		unsigned int pcnt, unused = 0;				\
+		const char *stack = K_THREAD_STACK_BUFFER(sym);		\
+		size_t size = K_THREAD_STACK_SIZEOF(sym);		\
+									\
+		unused = stack_unused_space_get(stack, size);		\
+									\
+		/* Calculate the real size reserved for the stack */	\
+		pcnt = ((size - unused) * 100) / size;			\
+									\
+		LOG_INF("%s (real size %u):\tunused %u\tusage %u / %u (%u %%)",\
```

Comment:
Btw, this "reel size:" stuff is a relic from the times when the stack meta-data was part of the stack itself (rather than a separate struct), resulting in a smaller call stack than the allocated buffer. Now days it's no different, as you can also so by you passing `size` twice as a parameter. So it should really be removed from all places printing this kind of info.

Reply:
removed